lib/tools.js

```diff
@@ -31,7 +31,7 @@ function genLog (level, hook) {
   if (!hook) return LOG
 
   return function hookWrappedLog (...args) {
-    hook.call(this, args, LOG)
+    hook.call(this, args, LOG, level)
```

Comment:
@mcollina this will retain a reference to the closure where `level` is defined. Do you foresee any optimization issues with this? Should the function be defined and a property attached to it so it can be `hook.call(this, args, LOG, this.level)`?

Reply:
This is ok for me, the closure context will be kept anyway.